### ISSUE TITLE
[Android][Expokit][Release] TypeError: _ExpoFontLoader.default.loadAsync is not a function

### ISSUE THREAD

**Issue by wpconsulate** [expo/expo#4131]
### Environment

Expo CLI 2.16.1 environment info:
    System:
      OS: macOS 10.14.3
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 11.9.0 - /usr/local/bin/node
      Yarn: 1.13.0 - /usr/local/bin/yarn
      npm: 6.5.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5314842
      Xcode: 10.2/10E125 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^32.0.0 => 32.0.6
      react: 16.5.0 => 16.5.0
      react-native: https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz => 0.57.1
      react-navigation: ^3.0.9 => 3.9.1
    npmGlobalPackages:
      expo-cli: 2.16.1

Android SDK target 28

### Steps to Reproduce

Upgraded old Project with Expokit and SDK 27 to Expo SDK 32 because of the https://forums.expo.io/t/app-removed-from-google-play-due-to-policy-violation-with-branch/21240

To upgrade I followed 
https://docs.expo.io/versions/latest/workflow/upgrading-expo-sdk-walkthrough/
and
https://docs.expo.io/versions/v32.0.0/expokit/expokit/#upgrading-expokit

App Works fine in Dev Mode. But in `release` mode 

```js
    try {
           await Font.loadAsync({
                'bold': require('./fonts/Roboto-Bold.ttf'),
                'regular': require('./fonts/Roboto-Medium.ttf')
            });
        }
       catch(e) {}
        finally {
            console.log(Font.isLoaded('bold'));
			console.log(Font.isLoaded('regular'));
			this.setState({ fontLoaded: true });
        }
```

Shows no font is being loaded. In Debug mode fonts are loading fine.

### Expected Behavior
Fonts should be loaded fine on the App.

### Actual Behavior
like stated above fonts aren't being loaded and app throws an error 

```
fontFamily "bold" is not a system font and has not been loaded through Font.loadAsync.

- If you intended to use a system font, make sure you typed the name correctly and that it is supported by your device operating system.

- If this is a custom font, be sure to load it with Font.loadAsync.
- node_modules/expo-font/build/Font.js:24:16 in processFontFamily
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:3512:38 in diffProperties
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:3417:6 in addNestedProperty
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:3417:6 in addNestedProperty
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:3554:8 in diffProperties
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:3969:29 in createInstance
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:11772:12 in completeWork
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:13902:10 in completeUnitOfWork
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:14117:30 in performUnitOfWork
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:14129:41 in workLoop
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:14226:15 in renderRoot
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:15193:17 in performWorkOnRoot
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:15090:24 in performWork
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:15047:14 in performSyncWork
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:14925:19 in requestWork
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:14711:16 in scheduleWork
- node_modules/react-native/Libraries/Renderer/oss/ReactNativeRenderer-dev.js:7700:17 in enqueueSetState
- node_modules/react/cjs/react.development.js:364:31 in setState
* app/index.js:73:42 in <unknown>
- node_modules/promise/setimmediate/core.js:37:14 in tryCallOne
- node_modules/promise/setimmediate/core.js:123:25 in <unknown>
- ... 8 more stack frames from framework internals

[Unhandled promise rejection: TypeError: _ExpoFontLoader.default.loadAsync is not a function. (In '_ExpoFontLoader.default.loadAsync(_getNativeFontName(name), asset.localUri)', '_ExpoFontLoader.default.loadAsync' is undefined)]
- node_modules/expo-font/build/Font.js:86:35 in _loadSingleFontAsync$
- node_modules/regenerator-runtime/runtime.js:62:44 in tryCatch
- ... 14 more stack frames from framework internals
```

**Comment by sjchmiela** [comment]
Hey, @wpconsulate! Could you try editing your `proguard-rules.pro` file so you're sure that the following fragment is present there?

https://github.com/expo/expo/blob/cfa00db583bc9ffda1765a4a480c00c67dfbabab/android/app/proguard-rules.pro#L31-L50

In SDK29 we've added universal modules to Expo, and for ProGuard not to strip out seemingly unnecessary methods (we call them using reflection) we need to instruct him not to do so. 🙂

**Comment by wpconsulate** [comment]
@sjchmiela I got that fixed. It was related to the `expo.gradle` file which wasn't copying the assets to correctly. 

so in that we needed to update the assets path in build to 
`def folderName = variant.name`
`        def targetName = folderName.capitalize()`
`        def assetsDir = file("$buildDir/intermediates/merged_assets/${folderName}/merge${targetName}Assets/out")`

and no info has been given in expo upgrade page related to this one. 

After this, I was able to build and assets ( images/ fonts ) worked fine.

Hope this helps anyone else.

thanks
